rest-tests/src/test/groovy/org/hawkular/metrics/rest/MetricsITest.groovy

```diff
@@ -150,4 +150,189 @@ class MetricsITest extends RESTTest {
+
+
+    response = hawkularMetrics.get(path: 'gauges/CG1/data', headers: [(tenantHeaderName): tenantId])
+    assertEquals(200, response.status)
+    assertEquals(
+        [
+            [timestamp: start.plusMinutes(1).millis, value: 9.589],
+            [timestamp: start.millis, value: 10.032]
+        ],
+        response.data
+    )
+
+    response = hawkularMetrics.get(path: 'availability/CA1/data', headers: [(tenantHeaderName): tenantId])
+    assertEquals(200, response.status)
+    assertEquals(
+        [
+            [timestamp: start.millis, value: "down"],
+            [timestamp: start.plusMinutes(1).millis, value: "up"]
+        ],
+        response.data
+    )
+  }
+
+  @Test
+  void addMixedDataMissingAvailabilities() {
+    String tenantId = nextTenantId()
+    DateTime start = DateTime.now().minusMinutes(10)
+
+    def response = hawkularMetrics.post(
+        path: "metrics/data",
+        headers: [(tenantHeaderName): tenantId],
+        body: [
+            gauges: [
+                [
+                    id: 'AG1',
+                    data:  [
+                        [timestamp: start.millis, value: 10.032],
+                        [timestamp: start.plusMinutes(1).millis, value: 9.589]
+                    ],
+                ]
+            ],
+            counters: [
+                [
+                    id: 'AC1',
+                    data: [
+                        [timestamp: start.millis, value: 10],
+                        [timestamp: start.plusMinutes(1).millis, value: 20]
+                    ]
+                ]
+            ]
+        ]
+    )
+    assertEquals(200, response.status)
+
+    response = hawkularMetrics.get(path: 'gauges/AG1/data', headers: [(tenantHeaderName): tenantId])
+    assertEquals(200, response.status)
+    assertEquals(
+        [
+            [timestamp: start.plusMinutes(1).millis, value: 9.589],
+            [timestamp: start.millis, value: 10.032]
+        ],
+        response.data
+    )
+
+    response = hawkularMetrics.get(path: 'counters/AC1/data', headers: [(tenantHeaderName): tenantId])
+    assertEquals(200, response.status)
+    assertEquals(
+        [
+            [timestamp: start.plusMinutes(1).millis, value: 20],
+            [timestamp: start.millis, value: 10]
+        ],
+        response.data
+    )
+  }
+
+  @Test
+  void addMixedDataEmptyRequestPayload() {
+    String tenantId = nextTenantId()
+
+    badPost( path: "metrics/data",
+        body: [],
+        headers: [(tenantHeaderName): tenantId]) { exception ->
+      // Missing type
+      assertEquals(400, exception.response.status)
+    }
+  }
+
+  @Test
+  void addMixedDataMissingData() {
+    String tenantId = nextTenantId()
+
+    def response = hawkularMetrics.post(
+        path: "metrics/data",
+        headers: [(tenantHeaderName): tenantId],
+        body: [
+            gauges: [
+            ],
+            counters: [
+            ]
+        ]
+    )
+    assertEquals(200, response.status)
```

Comment:
Shouldn't the status code be 400? Gauge handler replies with 400 when the list of metrics is empty.